src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -96,5 +86,244 @@ public bool NeedsRefresh()
+                                        reader.TokenType.ToString(),
+                                        $"{JsonTokenType.True}' or '{JsonTokenType.False}");
+                                }
+
+                                break;
+                            }
+
+                        case FeatureManagementConstants.ConditionsJsonPropertyName:
+                            {
+                                if (reader.Read() && reader.TokenType == JsonTokenType.StartObject)
+                                {
+                                    featureFlag.Conditions = ParseFeatureConditions(ref reader, settingKey);
+                                }
+                                else if (reader.TokenType != JsonTokenType.Null)
+                                {
+                                    throw CreateFeatureFlagFormatException(
+                                        FeatureManagementConstants.ConditionsJsonPropertyName,
+                                        settingKey,
+                                        reader.TokenType.ToString(),
+                                        JsonTokenType.StartObject.ToString());
+                                }
+
+                                break;
+                            }
+
+                        default:
+                            reader.Skip();
+
+                            break;
+                    }
+                }
+            }
+            catch (JsonException e)
+            {
+                throw new FormatException(settingKey, e);
+            }
+
+            return featureFlag;
+        }
+
+        private FeatureConditions ParseFeatureConditions(ref Utf8JsonReader reader, string settingKey)
+        {
+            var featureConditions = new FeatureConditions();
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    continue;
+                }
+
+                string conditionsPropertyName = reader.GetString();
+
+                switch (conditionsPropertyName)
+                {
+                    case FeatureManagementConstants.ClientFiltersJsonPropertyName:
+                        {
+                            if (reader.Read() && reader.TokenType == JsonTokenType.Null)
+                            {
+                                break;
+                            }
+                            else if (reader.TokenType == JsonTokenType.StartArray)
+                            {
+                                while (reader.Read() && reader.TokenType != JsonTokenType.EndArray)
+                                {
+                                    if (reader.TokenType == JsonTokenType.StartObject)
+                                    {
+                                        featureConditions.ClientFilters.Add(ParseClientFilter(ref reader, settingKey));
```

Comment:
Ok thanks, seems like maybe the latter makes more sense? Leave the parsing to the parse method and just handle the logic here

Reply:
The latter could get complicated. Imagine this json:
```json
{
    ...
    ...

    "client_filters": [
        { }, 
        {   
          "Name": "foo",
          "Parameters": { ... }   
        },
        { }
    ]

}
```

I think it might be better to add a new ClientFilter to the list of ClientFilters only when Name or Parameters is non-null. I believe this is what JsonSerializer does for us currently.

Reply:
Oh I see, I guess it makes sense this way if the serializer was already doing it, since we're still just doing parse logic by not adding empty filters. I was trying to compare it to the check for if `id` is null, but that's different.